test/integration/scheduler/scheduler_test.go

```diff
@@ -1267,12 +1267,14 @@ var _ = ginkgo.Describe("Scheduler", func() {
 
 			ginkgo.By("Finishing second workload for prod ClusterQueue")
 			util.FinishWorkloads(ctx, k8sClient, pWl2)
+			util.ExpectWorkloadToFinish(ctx, k8sClient, client.ObjectKeyFromObject(pWl2))
```

Comment:
Not sure I got how it can help.

Reply:
Yes, we set the status, but we do not wait for the status to be actually changed before we continue

Reply:
I have tested it in 3 ways:
1. repeat 10 times with no changes - couple failed bc of the issue
2. repeat 10 times with extended time, like 5* the current timeout - works all the time
3. repeat 10 times with this fix and timeout stays as it was in 1) - works all the time

Reply:
I see, so it basically splits the 5s timeout into two phases: wait for workload to finish, and wait for wl3 to reserve quota. I guess the test fails if both things don't happen within 5s currently.

Reply:
@mszadkow is my understanding above correct - just giving more time, or something more is going on? If this is not just about more time, then I also still don't understand why the fix helps.

Reply:
I don't think these changes help. I think it's just a coincidence.